plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchColumnHandle.java

```diff
@@ -13,28 +13,44 @@
  */
 package io.trino.plugin.opensearch;
 
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
 import io.trino.spi.connector.ColumnHandle;
 import io.trino.spi.type.Type;
 
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public record OpenSearchColumnHandle(
-        String name,
+        List<String> path,
         Type type,
         DecoderDescriptor decoderDescriptor,
-        boolean supportsPredicates)
+        boolean supportsPredicates,
+        String dereferencePath)
         implements ColumnHandle
 {
     public OpenSearchColumnHandle
     {
-        requireNonNull(name, "name is null");
+        path = ImmutableList.copyOf(requireNonNull(path, "path is null"));
```

Comment:
requireNonNull not needed for `path`. ImmutableList.copyOf() will throw if the argument is null.